### ISSUE TITLE
Pyzo freeze on autocomplete lookup loop 

### ISSUE THREAD

**Issue by axel-m-d** [pyzo/pyzo#944]
Hello,

I just discovered a small problem while writing code. The problem is caused by the developer, but maybe Pyzo can be resilient against it ?
It is regarding the autocomplete pop-up functionality when writing code.

When Pyzo is doing a lookup to find the methods and properties of the class you are subclassing AND if by mistake you named your subclass the same name as class : when typing `self.` pyzo will freeze (when typing the dot  `.`)
It only happens when creating a subclass of a class located in another file. When doing it in the same file, there is no problem, Pyzo is working fine

### Example :
You have 2 files in the same folder : `testAutoCompleteBaseClassFile.py` and `testAutoCompleteSubBaseClassFile.py`
#### testAutoCompleteBaseClassFile.py
```
class A(object):
    def __init__(self):
        self.a = 1
    def myMethodA(self):
        pass
```
#### testAutoCompleteSubBaseClassFile.py
```
from testAutoCompleteBaseClassFile import A

class A(A):
    def __init__(self):
        pass
```
When editing the second file `testAutoCompleteSubBaseClassFile.py` : 
In the `__init__` block if you replace :
`pass` by `self.a = 1` 
 Pyzo will freeze on the `self.`

It's a minor bug I think, but I took me some time to discover why Pyzo froze.
The error here is that the subclass declaration should be like `class B(A):` not `class A(A):` 

[testAutoCompleteBaseClassFile.py.txt](https://github.com/pyzo/pyzo/files/13757463/testAutoCompleteBaseClassFile.py.txt)
[testAutoCompleteSubClassFile.py.txt](https://github.com/pyzo/pyzo/files/13757464/testAutoCompleteSubClassFile.py.txt)

By the way : is there any way to Pyzo in "debug" mode if it exists ? I did not find any option to allow Pyzo to write logs on file and give more informations.
I tried the compiled version, and the version installed from pip in a conda environment, and the problem exists on both version.

My environment : 
```
Windows-10-10.0.19045-SP0
3.12.0 | packaged by Anaconda, Inc. | (main, Oct  2 2023, 17:20:38) [MSC v.1916 64 bit (AMD64)]
Pyzo 4.14.4 using PyQt6
```
I tried to find a possible code fix but this functionality seams a little bit complicated : maybe it's in the `pyzo\core\baseTextCtrl.py` but i'm not quite sure.

Kind regards,
Axel.

___

Also, I just want to tell you that Pyzo is an awesome software. I discovered it in engineering school in 2015, and it's my go-to IDE for my Python apps since ! 🙂

**Comment by bdieterm** [comment]
Hi @axel-m-d!
Thanks for narrowing down and reporting the problem.
I just created a patch for that, see pull request #945.

> By the way : is there any way to Pyzo in "debug" mode if it exists ? I did not find any option to allow Pyzo to write logs on file and give more informations.

I normally debug Pyzo by running Pyzo as a script inside Pyzo. This allows you to use the normal debugging features such as breakpoints, stack frame switching, etc..
Just download the Pyzo source code as a zip file, then run pyzolauncher.py from the downloaded source code inside your installed Pyzo IDE.
In this case I executed your code for reproducing the problem inside the "child" Pyzo and triggered the infinite loop.
Then I just interrupted the execution in the "parent" Pyzo (menu: Shell -> Interrupt [Ctrl+I]). This gave me the first stack trace and then I added print calls and breakpoints to find out more about the problem.
